qcommon/src/org/qcmg/common/model/PileupElement.java

```diff
@@ -73,12 +76,30 @@ public int getTotalCount() {
 	public boolean isFoundOnBothStrands() {
 		return forwardCount > 0 && reverseCount > 0;
 	}
-
+	@Override
 	public int compareTo(PileupElement o) {
 		// only interested in the total count for the purposes or ordering
 		return o.getTotalCount() - getTotalCount();
 	}
 	
+	//you need to implement the equals method thus overriding the default compareTo
+	@Override 
```

Comment:
this equals and hashcode methods look a little sparse.
Eclipse provides a decent implementation of these methods which will be a little more robust.
Right click in the class, go to `Source` -> `Generate hashCode and equals()`

Reply:
thanks for suggestion